cmake/modules/extensions.cmake

```diff
@@ -2602,10 +2613,73 @@ function(zephyr_get variable)
 
   if(GET_VAR_MERGE)
     list(REMOVE_DUPLICATES ${variable})
+    if(GET_VAR_REVERSE)
+      list(REVERSE ${variable})
+    endif()
```

Comment:
> Won't these cause conflict? If there is a duplicate in the list, e.g. A, B, C, D, E, A it will then become A, B, C, D, E and reversed to E, D, C, B, A so A will be applied last when someone might want it to be applied first

This is exactly as designed.
https://github.com/zephyrproject-rtos/zephyr/blob/e8fab54b6605aa6b8432318777a2a582d4219842/cmake/modules/extensions.cmake#L2499-L2502

And the order of precedence means first in list has highest precedence, in your example A has higher precedence than B (and therefore duplicated As can be removed).
When you reverse the list, for example because what is applied last wins, such as `EXTRA_CONF_FILE` where last file applied wins, then you still want `A` to win over `B`, hence `REVERSE` can be applied.

> when someone might want it to be applied first

That someone should only consider in which scope they want to add their setting.
Sysbuild, cache, env, snippets, current, and the scope decides what takes precedence, and the consumer of `zephyr_get()` knows whether first in list means highest priority, or if last in list wins.

> although specifying it twice would not be recommended

Depends on the use-case.
If `A` is default applied by a sample (current scope), and user wants A to have higher precedence, then user can add A to the cache using `-D<var>=A`, and it move higher up the list (and the current scope version gets removed).
Whether the consumer of the list wants to reverse the list doesn't change the fact that the cache variant of A still has moved up the list.